File: merge_two_linked_list.py

class Node:
	def __init__(self, val, next=None):
		self.val = val
		self.next = next

class Solution:
	def mergeTwoLists(self, a, b):
		if a == None or b == None:
			if a:
				return a
			elif b:
				return b
			else:
				return None
		elif a.val < b.val:
			c = Node(a.val)
			c.next = self.mergeTwoLists(a.next, b)
		else:
			c = Node(b.val)
			c.next = self.mergeTwoLists(b.next, a)
		return c

a = Node(1)
a.next = Node(3)
a.next.next = Node(5)


b = Node(2)
b.next = Node(4)
b.next.next = Node(6)

c = Solution().mergeTwoLists(a, b)

while c:
	print(c.val)
	c = c.next

File: merge_two_linked_list.cxx
#include <iostream>

using namespace std;

int main() {
	class Node {
		public:
			int val;
			Node *next;
			Node (int _val, Node *_next=0) {
				val = _val;
				next = _next;
			};
			Node *mergeTwoLists(Node *a, Node *b) {
				if (a && b) {
					if (a->val < b->val) {
						return new Node(a->val, mergeTwoLists(b, a->next));
					} else {
						return new Node(b->val, mergeTwoLists(a, b->next));
					};
				} else {
					if (a) {
						return a;
					} else {
						return b;
					};
				};
			};
	};
	Node node1(1);
	node1.next = new Node(3);
	node1.next->next = new Node(5);
	Node node2(2);
	
	node2.next = new Node(4);
	node2.next->next = new Node(6);
	Node *node3;
	node3 = node3->mergeTwoLists(&node1, &node2);

	while (node3) {
		cout << node3->val << " ";
		node3 = node3->next;
	};
};
